teachertool/src/transforms/setRubricName.ts

```diff
@@ -0,0 +1,18 @@
+import { stateAndDispatch } from "../state";
+import * as Actions from "../state/actions";
+
+export function setRubricName(name: string) {
+    const { state: teacherTool, dispatch } = stateAndDispatch();
+
+    const oldName = teacherTool.rubric.name;
+
+    if (oldName === name) {
+        return;
+    }
+
+    const newRubric = {
+        ...teacherTool.rubric,
+        name,
+    };
+    dispatch(Actions.setRubric(newRubric));
```

Comment:
May want to provide feedback here that the name was saved. Post a notification?

Reply:
Hmm, a notification feels a little too heavy-weight here to me. Ideally, I think the user would just feel like the name is always saved once they type it, similarly to how criteria adding/removing works, no real need for a "saved" notification. Maybe I should just save the name whenever the text is changed and debounce it a little, rather than requiring them to hit enter and/or click away from the Input?

Reply:
Probably fine as is for now. Let's see what kind of feedback we get.

Reply:
I did end up adding a debounced input to handle this instead...but I can change it back if you want :)

Reply:
(also considered just wrapping the onChange call in pxt.Util.debounce but I got worried it wouldn't play nicely with React re-rendering things and possibly resetting fields)